code/modules/mining/lavaland/weather.dm

```diff
@@ -0,0 +1,96 @@
+var/global/datum/weather/activeWeather
+
+#define STARTUP_STAGE 1
+#define MAIN_STAGE 2
+#define WIND_DOWN_STAGE 3
+#define END_STAGE 4
+
+/datum/weather
+	var/name = "storm"
+	var/start_up_time = 300 //30 seconds
+	var/start_up_message = "The wind begins to pick up."
+	var/duration = 120 //2 minutes
+	var/duration_message = "A storm has started!"
+	var/wind_down = 300 // 30 seconds
+	var/wind_down_message = "The storm is passing."
+
+	var/target_z = 1
+	var/exclude_walls = TRUE
+	var/area_type = /area/space
+	var/stage = STARTUP_STAGE
+
+
+	var/start_up_overlay = "lava"
+	var/duration_overlay = "lava"
+	var/purely_aesthetic = FALSE //If we just want gentle rain that doesn't hurt people
+
+/datum/weather/proc/weather_start_up()
+	if(activeWeather)
+		return
+	activeWeather = src
+	update_turfs()
+	for(var/mob/M in player_list)
+		if(M.z == target_z)
+			M << "[start_up_message]"
+
+	sleep(start_up_time)
+	stage = MAIN_STAGE
+	weather_main()
+
+
+/datum/weather/proc/weather_main()
+	update_turfs()
+	for(var/mob/M in player_list)
+		if(M.z == target_z)
+			M << "[duration_message]"
+	if(purely_aesthetic)
+		sleep(duration*10)
+	else  //Storm effects
+		for(var/i in 1 to duration-1)
+			for(var/mob/living/L in living_mob_list)	.
+				var/turf/Z = get_turf(L)
+				if(Z.weather == src)
+					storm_act(L)
+					sleep(10)
```

Comment:
This is too indented. This sleeps every for every mob affected. Remove two tabs and you'll get what you want.